### PR TITLE
Make DeepLabV3 + timm-mobilenetv3 scriptable.

### PR DESCRIPTION
**Pull request by hgaiser**
This makes at least DeepLabV3 and timm-mobilenetv3 scriptable. I figured this would be a good starting point to discuss if these changes are acceptable.

This PR fixes https://github.com/qubvel/segmentation_models.pytorch/issues/669

### CODE REVIEW

File: segmentation_models_pytorch/encoders/timm_mobilenetv3.py [hunk 1]
```diff
@@ -65,16 +68,14 @@ def get_stages(self):
...
-            ValueError("MobileNetV3 mode should be small or large, got {}".format(self._mode))
+            raise ValueError("MobileNetV3 mode should be small or large, got {}".format(self._mode))
 
     def forward(self, x):
-        stages = self.get_stages()
-
         features = []
-        for i in range(self._depth + 1):
-            x = stages[i](x)
+        for stage in self._stages:
```
Comment (qubvel) [inline segmentation_models_pytorch/encoders/timm_mobilenetv3.py:77]: If the depth is equal to 4, we do not need to pass the tensor through all 5 stages (this reduces time and space), that is why I iterate over depth

File: segmentation_models_pytorch/base/model.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def forward(self, x):
         self.check_input_shape(x)
 
         features = self.encoder(x)
-        decoder_output = self.decoder(*features)
+        decoder_output = self.decoder(features)
```
Comment (qubvel) [inline segmentation_models_pytorch/base/model.py:30]: This affects all decoders, unet, fpn, etc

File: segmentation_models_pytorch/encoders/timm_mobilenetv3.py (hunk 1 above)
Comment (hgaiser) [reply segmentation_models_pytorch/encoders/timm_mobilenetv3.py:77]: Ahh okay, thanks for the feedback. AFAIK there was an issue with looking up `self._stages[i]`, but I'll see if I can work around that.

File: segmentation_models_pytorch/base/model.py (hunk 2 above)
Comment (hgaiser) [reply segmentation_models_pytorch/base/model.py:30]: Yes sorry, I could've made that clearer that I was focusing on just one decoder / encoder pair to get your feedback on the changes. In the future I think it makes sense to add tests to test these conversions for different pairs as well.

File: segmentation_models_pytorch/encoders/timm_mobilenetv3.py (hunk 1 above)
Comment (hgaiser) [reply segmentation_models_pytorch/encoders/timm_mobilenetv3.py]: I changed the code to use `_depth` again. It's not my preferred way to code that behavior, but other methods didn't appear to be scriptable. Is this acceptable?

**Comment by hgaiser** [comment]
I'm currently stuck when converting a TorchScript model to ONNX. I am currently using the following script:

```python
import torch
import segmentation_models_pytorch as smp

model = smp.DeepLabV3Plus(encoder_name='mobilenet_v2',)
model.eval()

# Test the normal model.
input = torch.randn((1, 3, 224, 224))
torch_output = model(input)

# Convert to TorchScript and get its output.
model = torch.jit.script(model)
torchscript_output = model(input)

# Assert that the results are similar.
torch.testing.assert_allclose(torch_output, torchscript_output)

# Export to ONNX.
torch.onnx.export(
    model,
    input,
    'model.onnx',
)

# Test the model using ONNX.
import onnx
model = onnx.load('./model.onnx')
onnx.checker.check_model(model)

# Load the model using onnxruntime.
import onnxruntime
onnxruntime.InferenceSession('./model.onnx')
```

Everything works, *except* for loading the ONNX model using onnxruntime. I get the following error:

```
onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from ./model.onnx failed:Node (Concat_165) Op (Co
ncat) [ShapeInferenceError] All inputs to Concat must have same rank. Input 1 has rank 2 != 1
```

I've tracked it down to [this](https://github.com/qubvel/segmentation_models.pytorch/blob/master/segmentation_models_pytorch/decoders/deeplabv3/decoder.py#L152) line, but no clue yet why that is. In the unlikely case that this helps someone, here ya go :p

I will proceed and see if I can figure this out..

**Comment by hgaiser** [comment]
I am very slowly progressing, but it definitely is the interpolate function that is not converting properly to ONNX. I'll see if I can make a minimum working example and submit a bug report with torch.

**Comment by hgaiser** [comment]
I have submitted a bug report to pytorch regarding the progress on this subject:

https://github.com/pytorch/pytorch/issues/87864

**Comment by hgaiser** [comment]
I'll close this PR for now. There's a few bugs upstream that should be resolved first. I might pick this up again in the future. For now I'll stick to tracing :).

**Pull request closed without merging**